include/nanospline/PatchBase.h

```diff
@@ -172,9 +172,9 @@ class PatchBase
         UVPoint uv(min_u, min_v);
         Scalar min_dist = std::numeric_limits<Scalar>::max();
         for (int i = 0; i <= num_samples; i++) {
-            const Scalar u = i * (max_u - min_u) / num_samples + min_u;
+            const Scalar u = (i == num_samples) ? max_u : i * (max_u - min_u) / num_samples + min_u;
```

Comment:
When i == num_samples, these should be the same value, right? Is there a floating point issue?

Reply:
Yes, this is to avoid floating point error because it may trigger the assertion `u <= max_u`.